frontend/app/invoices/page.tsx

```
@@ -119,31 +210,214 @@ export default function InvoicesPage() {
+  const handleInvoiceAction = (actionId: string, invoices: Invoice[]) => {
+    switch (actionId) {
+      case "approve":
+        setOpenModal("approve");
+        break;
+      case "reject":
+        setOpenModal("reject");
+        break;
+      case "delete": {
+        // Ensure selected invoices are marked for deletion
+        const invoiceIds = invoices.map((inv) => inv.id);
+        table.setRowSelection(invoiceIds.reduce((acc, id) => ({ ...acc, [id]: true }), {}));
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)